graphql_compiler/tests/test_input_data.py

```diff
@@ -2465,3 +2465,30 @@ def complex_nested_optionals():
         expected_output_metadata=expected_output_metadata,
         expected_input_metadata=expected_input_metadata,
         type_equivalence_hints=None)
+
+
+def recursive_field_type_is_subtype_of_parent_field():
```

Comment:
When you add a test input in `test_input_data.py`, you must not only add a test case for it in `test_compiler.py` but also add one in `test_ir_generation.py`. I suspect tests will fail on this.

Reply:
yup both are in this PR :)